### PR TITLE
Clarify meaning of `task.id` versus `task.index`

### LINKED ISSUE #5482: Clarify that `task.index` is specific to the process
We should clarify the `task.index` is specific to the process, because the documentation implies it is the index of the task of the entire pipeline:

> The task index (corresponds to task_id in the [execution trace](https://www.nextflow.io/docs/latest/reports.html#trace-report)).

_Originally posted by @adamrtalbot in https://github.com/nextflow-io/nextflow/issues/5481#issuecomment-2462374081_

### PR DESCRIPTION
**Pull request by bentsherman**
Close #5482

### CODE REVIEW

File: docs/reference/process.md [hunk 1]
```diff
@@ -19,8 +19,11 @@ The following task properties are defined in the process body:
 : *Available only in `exec:` blocks*
 : The task unique hash ID.
 
+`task.id`
+: The pipeline-level task index (corresponds to `task_id` in the {ref}`execution trace <trace-report>`).
```
Comment (christopher-hakkaart) [inline docs/reference/process.md:23]:
```suggestion
: The pipeline-level task index. Corresponds to `task_id` in the {ref}`execution trace <trace-report>`.
```

**Review by christopher-hakkaart — APPROVED** [review]
It's best to limit (brackets) where possible. Non blocking.

**Comment by bentsherman** [comment]
@pditommaso for your consideration

File: docs/reference/process.md [hunk 2]
```diff
@@ -19,8 +19,11 @@ The following task properties are defined in the process body:
 : *Available only in `exec:` blocks*
 : The task unique hash ID.
 
+`task.id`
+: The pipeline-level task index. Corresponds to `task_id` in the {ref}`execution trace <trace-report>`.
+
 `task.index`
-: The task index (corresponds to `task_id` in the {ref}`execution trace <trace-report>`).
+: The process-level task index.
```
Comment (pditommaso) [inline docs/reference/process.md:27]: why removing 

>  (corresponds to `task_id` in the {ref}`execution trace <trace-report>`).

Comment (bentsherman) [reply docs/reference/process.md:27]: Look one line above, it was actually `task.id` the whole time 😅

Comment (pditommaso) [reply docs/reference/process.md:27]: Ouch!

**Review by pditommaso — APPROVED** [review]

**Pull request merged**